packages/main-api/src/controllers/v1/contents.rs

```diff
@@ -182,6 +182,21 @@ impl ContentController {
 
         let mut tx = self.pool.begin().await?;
 
+        // Check if the content has already been minted by this user
+        let already_minted = sqlx::query_scalar::<_, bool>(
+            "SELECT EXISTS(SELECT 1 FROM content_downloads WHERE user_id = $1 AND content_id = $2)",
+        )
+        .bind(user_id)
+        .bind(content_id)
+        .fetch_one(&mut *tx)
```

Comment:
```suggestion
        .fetch_optional(&mut *tx)
```

In transaction, fetch_one can cause dead_lock. So we should use fetch_optional